lib/src/components/NxGlobalSidebar/NxGlobalSidebar.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { FunctionComponent } from 'react';
+import classnames from 'classnames';
+
+import NxButton from '../NxButton/NxButton';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { useRandomId } from '../../util/idUtil';
+import { Props, propTypes } from './types';
+
+import './NxGlobalSidebar.scss';
+
+const NxGlobalSidebar: FunctionComponent<Props> = function NxGlobalSidebar(props) {
+  const {
+    isOpen,
+    children,
+    className,
+    toggleOpenIcon,
+    toggleCloseIcon,
+    onToggleClick,
+    logoImg,
+    logoAltText,
+    logoLink
+  } = props;
+
+  const id = useRandomId('nx-global-sidebar');
+
+  const classes = classnames(className, 'nx-global-sidebar', {
+    'open': isOpen,
+    'closed': !isOpen
+  });
+
+  const toggleButtonIcon = isOpen ? toggleOpenIcon : toggleCloseIcon;
+  const toggleAriaLabel = isOpen ? 'Collapse Sidebar' : 'Expand Sidebar';
+  const toggleButton = (
+    <NxButton aria-label={toggleAriaLabel}
+              aria-expanded={isOpen}
+              aria-controls={id}
+              variant="icon-only"
+              onClick={onToggleClick}
+              className="nx-global-sidebar__toggle">
+      <NxFontAwesomeIcon icon={toggleButtonIcon} fixedWidth />
+    </NxButton>
+  );
+
+  const logo = <img src={logoImg} alt={logoAltText} className="nx-global-sidebar__logo" />;
+
+  return (
+    <aside className={classes} id={id}>
```

Comment:
Curious why not just `id="nx-global-sidebar"`?

Reply:
> I was instructed to use the useRandomId function. I think it decreases the chance of id's colliding?

A hardcoded id _could_ be ok here since a real page should only have one of these sidebars.  At the same time, even in this case you might run into problems, for instance when writing documentation pages you might end up with multiple sidebar examples on the page.

Reply:
> > I was instructed to use the useRandomId function. I think it decreases the chance of id's colliding?
> 
> A hardcoded id _could_ be ok here since a real page should only have one of these sidebars. At the same time, even in this case you might run into problems, for instance when writing documentation pages you might end up with multiple sidebar examples on the page.

Ah, I see "nx-global-sidebar" is used as prefix. I confused the arg with explicitId.